_docs/concepts/auth.md

```diff
@@ -0,0 +1,138 @@
+
+The page explains Auth's role and general architecture.
+
+
+## Istio Auth Concept
+
+*Status: Draft*
+
+*Date: April, 2017*
+
+
+[TOC]
+
+
+### Overview {#overview}
+
+Istio Auth aims at enhancing the security of microservices and their communication without requiring service code changes. It is responsible for:
+
+
+
+*   Presenting a strong identity that represents the role of the service to enable interoperability across clusters and clouds.
+*   Securing the service to service communication and the end-user to service communication.
+*   Providing a key management system to automate key/cert generation, distribution, rotation, and revocation.
+*   Upcoming features:
+    *   fine-grained authorization and auditing to control and monitor who accesses your services, apis, or resources, 
+    *   powerful authorization mechanisms: [ABAC](https://en.wikipedia.org/wiki/Attribute-Based_Access_Control), [RBAC](https://en.wikipedia.org/wiki/Role-based_access_control), Authorization hooks.
+
+### Architecture {#architecture}
+
+Figure 1 shows the Istio Auth architecture, which includes three important components: identity, key management, and communication security. This diagram describes how Istio Auth is used to secure the service-to-service communication between service A running as the service account "foo" and service B running as the service account "bar".
+
+<img style="display:block;width:60%;margin:auto;" src="./img/auth/auth.svg"
+alt="Flow of traffic." />
+
+### Components {#components}
+
+#### Identity {#identity}
+
+Istio Auth uses [Kubernetes service accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) to identify who runs the service:
+
+
+
+*   Service account is **the identity (or role) the workload runs as**, which represents the privilege of the workload. For the system requiring strong security, the amount of privilege for a workload should not be identified by a random string (i.e., service name, label, etc), or by the binary that is deployed.
+    *   For example, let's say we have a workload pulling data from a multi-tenant database. If Alice ran this workload, she will be able to pull a different set of data than if Bob ran this workload. 
+*   Service account enables powerful security policy by offering the flexibility to identify a machine, a user, a workload, or a group of workloads (different workloads can run as the same service account).
+*   The service account a workload runs as won't change during the lifetime of the workload.
+*   Service account uniqueness can be ensured with domain name constraint
+
+#### Communication Security {#communication-security}
+
+Service-to-service communication is tunneled through the client side [Envoy](https://lyft.github.io/envoy/) and the server side Envoy. The end-to-end communication is secured by:
+
+
+
+*   Local TCP connections between the service and Envoy
+    *   We are looking into using unix domain socket for stronger security
```

Comment:
I'd leave out the "we are looking into" line as it's in the Future Work section.